overlord/managers_test.go

```diff
@@ -1955,6 +1956,336 @@ type: kernel`
 	c.Check(restarting, Equals, true)
 }
 
+func (s *mgrsSuite) TestInstallKernelSnap20UpdatesBootloaderEnv(c *C) {
+	bloader := bootloadertest.Mock("mock", c.MkDir())
+	bootloader.Force(bloader)
+	defer bootloader.Force(nil)
+
+	// the reboot var we care about for UC20 is kernel_status
+	bloader.RebootStatusVar = "kernel_status"
```

Comment:
this feels like the wrong level of detail to control here like this.  I suggest either:

* pass another param to Mock but that might be too annoying, too much churn
* a new bootloader.MockUC20   (but this might give the impression that is a completely different thing ?)
* a pattern that we don't use often but could work here,  do

  ```
  bloader := bootloader.Mock("mock", c.MkDir).UC20RunMode()
  ```

where UC20RunMode  returns its target and tweaks all the bits to make it behave in the right way for a UC20 run mode bootloader?

Reply:
Well we do already have `boottest.MockDevice` and `boottest.MockUC20Device` which do very similar things to what we do here, just return the same thing with a little internal tweak on it, so I would say let's go with that, but since the only real difference here is in the bootenv var used when emulating a reboot, I'd be afraid of folks thinking that there's more that needs to change, when really the only place you would need to use `bootloader.MockUC20` instead of `bootloader.Mock` would be in these tests specifically.

For these reasons, I think I will go with the 3rd alternative, but rename the method to be `bloader := bootloader.Mock("mock",c.Mkdir),UC20RebootMode()` because that's what is really being changed here.